### ISSUE TITLE
Bring your own `process.env`

### ISSUE THREAD

**Issue by moeriki** [motdotla/dotenv-expand#55]
`ignoreProcessEnv: true` is great, though undocumented 🙈, because we can prevent mutating `process.env`.

Sadly this option has the effect that we cannot interpolate variables across different files. Something that is quite convenient in a dotenv setup like `.env.local`, …, `.env`.

Ideally this option would be `useProcessEnv: boolean | object` instead of `ignoreProcessEnv: boolean` so we can loop over the seperate dotenv files without modifying `process.env` and without loosing the ability to refer variables across the files.

```ini
# .env.local
NAME="Dieter"
```

```ini
# .env
ADMIN="${NAME}"
```

```js
const dotenv = require('dotenv');
const expand = require('dotenv-expand');

const env = {};
expand(dotenv.config({ path: '.env.local' }), { ignoreProcessEnv: true, useProcessEnv: env });
expand(dotenv.config({ path: '.env' }), { ignoreProcessEnv: true, useProcessEnv: env });

// Expected
// assert.ok(env.ADMIN === 'Dieter');
```

**Comment by motdotla** [comment]
Yes, I see what you mean. I'll think on this.. maybe a tighter integration/hook with the `dotenv` module.

In other news, `ignoreProcessEnv` is now partially documented: https://github.com/motdotla/dotenv-expand#ignoreprocessenv

**Comment by WoodyWoodsta** [comment]
I think there is a more fundamental problem with this option, and probably why it wasn't documented in the first place. It does too much.

Not only does it not write to `process.env`, as the documentation suggests, but it also does not read in values from `process.env`. This means it cuts out a whole host of use-cases.

I think the original suggestion was a good one, but doesn't completely solve the underlying problem, which is that there should be a way to say to the module: "Read in values from `process.env`, but don't write them out to it". This can be achieved by making the type of `ignoreProcessEnv: boolean | 'read-only'`. When set to `read-only`, expand will read from `process.env` but not write to it.

Then, consider also adding an option to inject your own process.env, however, this should be achievable using `Object.assign` or spreading in between each file load.

**Comment by WoodyWoodsta** [comment]
@motdotla Any thoughts on the above? Would be great to get some direction on whether this should live in the `dotenv` module or out here for now.

**Comment by motdotla** [comment]
you can now bring your own `processEnv`. will be released as `v11.0.0` shortly.

```
const myObject = {}
const dotenv = {
  processEnv: myObject,
  parsed: {
    SHOULD_NOT_EXIST: 'testing'
  }
}
const obj = dotenvExpand.expand(dotenv).parsed

console.log(obj.SHOULD_NOT_EXIST) // testing
console.log(myObject.SHOULD_NOT_EXIST) // testing
console.log(process.env.SHOULD_NOT_EXIST) // undefined
```

https://github.com/motdotla/dotenv-expand?tab=readme-ov-file#processenv
